sql/src/main/java/org/vertexium/sql/SqlStreamingPropertyTable.java

```diff
@@ -43,14 +46,20 @@ public StreamingPropertyValueRef put(String elementId, String key, String name,
                                 "insert into %s (%s, %s, %s, %s) values (?, ?, ?, ?)",
                                 tableName, KEY_COLUMN_NAME, VALUE_COLUMN_NAME, VALUE_TYPE_COLUMN_NAME,
                                 VALUE_LENGTH_COLUMN_NAME),
-                        id, value.getInputStream(), value.getValueType().getName(), value.getLength());
+                        id, streamAndLength.inputStream, value.getValueType().getName(), streamAndLength.length);
             } else {
                 handle.execute(String.format(
                                 "update %s set %s = ?, %s = ? where %s = ?",
                                 tableName, VALUE_COLUMN_NAME, VALUE_LENGTH_COLUMN_NAME, KEY_COLUMN_NAME),
-                        value.getInputStream(), value.getLength(), id);
+                        streamAndLength.inputStream, streamAndLength.length, id);
             }
             return new SqlStreamingPropertyValueRef(value, elementId, key, name, visibility, timestamp);
+        } finally {
+            try {
+                streamAndLength.inputStream.close();
+            } catch (IOException ex) {
+                Throwables.propagate(ex);
```

Comment:
can you throw a `VertexiumException` here instead with a message like "could not close input stream". I'm not a fan of `Throwables.propagate` since you end up getting a raw `RuntimeException`